### PR TITLE
ref(ui) Convert InviteMemberModalRow control to new select

### PR DESCRIPTION
**Pull request by markstory**
This was the most complex select control as it has customized selected value items and the styling API has changed from v2 to v3.

New select value state

Screen Shot 2021-02-18 at 12 09 08 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/components/forms/selectControl.tsx [hunk 1]
```diff
@@ -225,7 +225,7 @@ function SelectControl<OptionType extends GeneralSelectValue = GeneralSelectValu
       ...provided,
       color: '#007eff',
       backgroundColor: '#ebf5ff',
-      borderRadius: '2px',
+      borderRadius: 2,
```
Comment (k-fish) [inline src/sentry/static/sentry/app/components/forms/selectControl.tsx:228]: I'm assuming this is intentional, but shouldn't border radius be in px?

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx [hunk 2]
```diff
@@ -24,94 +27,182 @@ type Props = {
...
+          <RoleSelectControl
+            data-test-id="select-role"
+            disabled={disabled}
+            value={role}
+            roles={roleOptions}
+            disableUnallowed={roleDisabledUnallowed}
+            onChange={onChangeRole}
+          />
+        </div>
+        <div>
```
Comment (k-fish) [inline src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx:181]: Are these extra divs necessary for layout? Or can we remove them?

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx [hunk 3]
```diff
@@ -24,94 +27,182 @@ type Props = {
...
+            }}
+            options={mapToOptions(emails)}
+            onBlur={(e: React.ChangeEvent<HTMLInputElement>) =>
+              e.target.value &&
+              onChangeEmails([
+                ...mapToOptions(emails),
+                {label: e.target.value, value: e.target.value},
+              ])
+            }
+            styles={{
```
Comment (k-fish) [inline src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx:116]: nit, but these inline styles are pretty noisy, maybe move them to their own function?

**Review by k-fish — APPROVED** [review]

File: src/sentry/static/sentry/app/components/forms/selectControl.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/forms/selectControl.tsx]: Whoops this is a remnant from when I was debugging the reversed style merging.

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx]: Looks like they aren't needed. I'll remove them.

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/index.tsx [hunk 4]
```diff
@@ -358,12 +358,7 @@ class InviteMembersModal extends AsyncComponent<Props, State> {
             teamOptions={allTeams}
             inviteStatus={inviteStatus}
             onRemove={() => this.removeInviteRow(i)}
-            onChangeEmails={opts =>
-              this.setEmails(
-                opts.map(v => v.value),
-                i
-              )
-            }
+            onChangeEmails={opts => this.setEmails(opts ? opts.map(v => v.value) : [], i)}
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/modals/inviteMembersModal/index.tsx:361]:
```suggestion
            onChangeEmails={opts => this.setEmails(opts?.map(v => v.value) ?? [], i)}
```

🤷‍♀️

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx [hunk 5]
```diff
@@ -24,94 +27,185 @@ type Props = {
...
+function getStyles(theme: Theme, inviteStatus: Props['inviteStatus']) {
+  return {
+    multiValue: (
+      provided: React.CSSProperties,
+      {data}: MultiValueProps<SelectOption>
+    ) => {
+      const status = inviteStatus[data.value];
+      return {
+        ...provided,
+        ...(status && status.error
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx:173]:
```suggestion
        ...(status?.error
```

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx [hunk 6]
```diff
@@ -24,94 +27,185 @@ type Props = {
...
+    },
+    multiValueLabel: (
+      provided: React.CSSProperties,
+      {data}: MultiValueProps<SelectOption>
+    ) => {
+      const status = inviteStatus[data.value];
+      return {
+        ...provided,
+        pointerEvents: 'all',
+        ...(status && status.error ? {color: theme.red300} : {}),
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx:190]:
```suggestion
        ...(status?.error ? {color: theme.red300} : {}),
```

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx [hunk 7]
```diff
@@ -24,94 +27,185 @@ type Props = {
...
+      };
+    },
+    multiValueRemove: (
+      provided: React.CSSProperties,
+      {data}: MultiValueProps<SelectOption>
+    ) => {
+      const status = inviteStatus[data.value];
+      return {
+        ...provided,
+        ...(status && status.error
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/modals/inviteMembersModal/inviteRowControl.tsx:200]:
```suggestion
        ...( status?.error
```

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**